### DISCUSSION TITLE
请问使用datastream方式如何同步多张表呢

### DISCUSSION THREAD

**Discussion by 18772247265** [apache/flink-cdc#413] [Q&A]
cdc使用dataStream，读取的多个表的binlog，目标表对应的也是多个表，这种情况下，如何做到同步更新；
使用sql的方式比较简单，但是只能对单表操作，请问这个需求改怎么解决

**Comment by leonardBang** [comment]
看起来像是整库同步的需求？

**Reply by 18772247265** [reply]
嗯，是这个意思，这种需求要怎么实现呢

**Comment by leonardBang** [comment]
SQL上的整库同步还在规划中，你可以使用datastream API, 自己解析，目前不少用户已经在用了

**Reply by 18772247265** [reply]
嗯嗯，好的，非常感谢

**Reply by Memory5210** [reply]
问一下，startupOptions这个你们选择的什么？怎么保证程序重启之后接着消费，同时保证不重复和不丢失

**Reply by Memory5210** [reply]
DataStream的形式，不是SQL

**Reply by zjb-it** [reply]
> 问一下，startupOptions这个你们选择的什么？怎么保证程序重启之后接着消费，同时保证不重复和不丢失

您好，问题方案有了没

**Reply by Memory5210** [reply]
默认的initial就行，做checkpoint

**Reply by zjb-it** [reply]
> 默认的initial就行，做checkpoint

您好，checkpoint做了  startupOptions用的initial，项目重启以后还是全部拉取了一遍数据，麻烦问下，还需要其它设置吗：
  env.enableCheckpointing(1000, CheckpointingMode.EXACTLY_ONCE);
 env.getCheckpointConfig().setCheckpointStorage("file:///C:\\Users\\Desktop\\");

**Reply by Memory5210** [reply]
> > 默认的initial就行，做checkpoint
> 
> 您好，checkpoint做了 startupOptions用的initial，项目重启以后还是全部拉取了一遍数据，麻烦问下，还需要其它设置吗： env.enableCheckpointing(1000, CheckpointingMode.EXACTLY_ONCE); env.getCheckpointConfig().setCheckpointStorage("file:///C:\Users\Desktop\");

下次启动要从checkpoint点或者savepoint点启动，不然做检查点的意义在哪，这个具体的操作指令可以直接在flink官网上查看，很容易找到。

**Comment by 18772247265** [comment]
默认不设置的话，就是先全量，再增量。如果你项目重启了，你启动的时候要把checkpoint路径带上




------------------&nbsp;原始邮件&nbsp;------------------
发件人:                                                                                                                        "ververica/flink-cdc-connectors"                                                                                    ***@***.***&gt;;
发送时间:&nbsp;2022年2月16日(星期三) 中午11:17
***@***.***&gt;;
***@***.******@***.***&gt;;
主题:&nbsp;Re: [ververica/flink-cdc-connectors] 请问使用datastream方式如何同步多张表呢 (Discussion #413)





   
默认的initial就行，做checkpoint
  
您好，checkpoint做了 startupOptions用的initial，项目重启以后还是全部拉取了一遍数据，麻烦问下，还需要其它设置吗： env.enableCheckpointing(1000, CheckpointingMode.EXACTLY_ONCE); env.getCheckpointConfig().setCheckpointStorage("file:///C:\Users\Desktop");
  
下次启动要从checkpoint点或者savepoint点启动，不然做检查点的意义在哪，这个具体的操作指令可以直接在flink官网上查看，很容易找到。
 
—
Reply to this email directly, view it on GitHub, or unsubscribe.
Triage notifications on the go with GitHub Mobile for iOS or Android.
